drivers/clock_control/clock_control_litex.c

```diff
@@ -0,0 +1,1791 @@
+	0b1111111111010101111011111010010000000001,
+	0b1111111111010100010111111010010000000001,
+	0b1111111111010100010111111010010000000001,
+	0b1111111111010010110011111010010000000001,
+	0b1111111111010010110011111010010000000001,
+	0b1111111111010010110011111010010000000001,
+	0b1111111111010001001111111010010000000001,
+	0b1111111111010001001111111010010000000001,
+	0b1111111111010001001111111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001,
+	0b1111111111001111101011111010010000000001
+};
+/* End of copied code */
+
+/* Helper function for filter lookup table */
+static uint32_t litex_clk_lookup_filter(uint32_t glob_mul)
+{
+	return litex_clk_filter_table[glob_mul - 1];
+}
+
+/* Helper function for lock lookup table */
+static uint64_t litex_clk_lookup_lock(uint32_t glob_mul)
+{
+	return litex_clk_lock_table[glob_mul - 1];
+}
+
+static inline void litex_clk_set_reg(uint32_t reg, uint32_t val)
+{
+	litex_write((uint32_t *)drp[reg].addr, drp[reg].size, val);
+}
+
+static inline uint32_t litex_clk_get_reg(uint32_t reg)
+{
+	return litex_read((uint32_t *)drp[reg].addr, drp[reg].size);
+}
+
+static inline void litex_clk_assert_reg(uint32_t reg)
+{
+	int assert = (1 << (drp[reg].size * BITS_PER_BYTE)) - 1;
+
+	litex_clk_set_reg(reg, assert);
+}
+
+static inline void litex_clk_deassert_reg(uint32_t reg)
+{
+	litex_clk_set_reg(reg, ZERO_REG);
+}
+
+static int litex_clk_wait(uint32_t reg)
+{
+	uint32_t timeout;
+
+	__ASSERT(reg == DRP_LOCKED || reg == DRP_DRDY, "Wrong reg value!");
+
+	if (reg == DRP_LOCKED) {
+		timeout = ldev->timeout.lock;
+	} else {
+		timeout = ldev->timeout.drdy;
+	}
+	/*Waiting for signal to assert in reg*/
+	while (!litex_clk_get_reg(reg) && timeout) {
+		timeout--;
+		k_sleep(K_MSEC(1));
+	}
+	if (timeout == 0) {
+		LOG_WRN("Wait timeout");
```

Comment:
can we make this more descriptive?

Reply:
@kgugala: Changed the message - is the new version ok?

Reply:
yep, thanks